CMakeLists.txt

```diff
@@ -285,35 +292,75 @@ else ((NOT HSA_HEADER) OR (NOT HSA_LIBRARY))
 endif ((NOT HSA_HEADER) OR (NOT HSA_LIBRARY))
 
 if (HAS_ROCM EQUAL 0)
-  MESSAGE(FATAL_ERROR "ROCm is NOT available on the system!")
+  message(FATAL_ERROR "ROCm is NOT available on the system!")
 endif (HAS_ROCM EQUAL 0)
 
 #################
 # Detect AMDGPU backend for native codegen
 #################
 
-set(HSA_AMDGPU_GPU_TARGET "gfx803" CACHE STRING "Target AMDGPU ISA version")
+set(default_target "gfx803")
+
+# Attempt to detect any available HSA Agents, picking the highest ISA they
```

Comment:
@AlexVlx I'm for @scchan 's idea to fix `default_target` to `gfx803` without invoking `rocm_agent_enumerator` at CMake time. This could help decouple the configuration used in the build machine from the machines being deployed.